atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/QosCassandraClient.java

```diff
@@ -118,56 +115,33 @@ private long getApproximateWriteByteCount(Map<ByteBuffer, Map<String, List<Mutat
     public ColumnOrSuperColumn get(TableReference tableReference, ByteBuffer key, byte[] column,
             ConsistencyLevel consistency_level)
             throws InvalidRequestException, NotFoundException, UnavailableException, TimedOutException, TException {
-        qosClient.checkLimit();
-
-        ColumnOrSuperColumn result = client.get(tableReference, key, column, consistency_level);
-        recordBytesRead(() -> ThriftObjectSizeUtils.getColumnOrSuperColumnSize(result));
-        return result;
+        return qosClient.executeRead(
+                () -> DEFAULT_ESTIMATED_READ_BYTES,
+                () -> client.get(tableReference, key, column, consistency_level),
+                ThriftObjectSizeUtils::getColumnOrSuperColumnSize);
     }
 
     @Override
     public CASResult cas(TableReference tableReference, ByteBuffer key, List<Column> expected, List<Column> updates,
             ConsistencyLevel serial_consistency_level, ConsistencyLevel commit_consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        qosClient.checkLimit();
-
-        CASResult result = client.cas(tableReference, key, expected, updates, serial_consistency_level,
+        // TODO(nziebart): should this be considered as a write or do we need to treat is as both read and write?
+        return client.cas(tableReference, key, expected, updates, serial_consistency_level,
                 commit_consistency_level);
-        recordBytesWritten(() -> getCollectionSize(updates, ThriftObjectSizeUtils::getColumnSize));
-        recordBytesRead(() -> getCollectionSize(updates, ThriftObjectSizeUtils::getColumnSize));
-        return result;
     }
 
     @Override
     public CqlResult execute_cql3_query(CqlQuery cqlQuery, Compression compression, ConsistencyLevel consistency)
             throws InvalidRequestException, UnavailableException, TimedOutException, SchemaDisagreementException,
             TException {
-        qosClient.checkLimit();
-
-        CqlResult cqlResult = client.execute_cql3_query(cqlQuery, compression, consistency);
-        recordBytesRead(() -> ThriftObjectSizeUtils.getCqlResultSize(cqlResult));
-        return cqlResult;
-    }
-
-    private void recordBytesRead(Supplier<Long> numBytesRead) {
-        try {
-            qosMetrics.updateReadCount();
-            qosMetrics.updateBytesRead(numBytesRead.get());
-        } catch (Exception e) {
-            log.warn("Encountered an exception when recording read metrics.", e);
-        }
-    }
 
-    private void recordBytesWritten(Supplier<Long> numBytesWritten) {
-        try {
-            qosMetrics.updateWriteCount();
-            qosMetrics.updateBytesWritten(numBytesWritten.get());
-        } catch (Exception e) {
-            log.warn("Encountered an exception when recording write metrics.", e);
-        }
+        return qosClient.executeRead(
```

Comment:
nit, should probably javadoc at some point that we treat this as if it's a read.